lib/compat/wordpress-6.7/rest-api.php

```diff
@@ -0,0 +1,35 @@
+<?php
+/**
+ * PHP and WordPress configuration compatibility functions for the Gutenberg
+ * editor plugin changes related to REST API.
+ *
+ * @package gutenberg
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	die( 'Silence is golden.' );
+}
+
+if ( ! function_exists( 'gutenberg_register_wp_rest_post_types_meta_fields' ) ) {
+	/**
+	 * Adds `template` and `template_lock` fields to WP_REST_Post_Types_Controller class.
+	 */
+	function gutenberg_register_wp_rest_post_types_meta_fields() {
+		register_rest_field(
+			'type',
+			'meta',
+			array(
+				'get_callback' => function ( $item ) {
+					return get_registered_meta_keys( $item['slug'] );
```

Comment:
According to the [get_registered_meta_keys function](https://developer.wordpress.org/reference/functions/get_registered_meta_keys/), I believe we have to pass `post` as the `object_type` and `$item['slug']` as the `object_subtype`. The way it is doesn't add the fields to other post types like pages or custom post types.

Reply:
The `meta` seems like a very general attribute name. There might be a collision.

We should also be careful about what data we expose publicly.

Reply:
> The meta seems like a very general attribute name. There might be a collision.

It seems `meta` is used in other endpoints (like `posts`) to show the meta fields. I feel it would be good for consistency but you are right it could conflict with other plugins extending the API. Any suggestion?

> We should also be careful about what data we expose publicly.

Totally agree on this. I would probably start showing only the required properties and expand them based on feedback.

Reply:
We can resolve this conversation, as we used the approach they mention before.